models/dashboards/pevr/pbi_tables/pevr_indicateur_des.sql

```diff
@@ -1,119 +1,149 @@
-    ),
-
-    agg_dip as (
-        select
-            '1.1.1.1' as id_indicateur,
-            annee_scolaire,
-            school_friendly_name,
-            genre,
-            plan_interv_ehdaa,
-            population,
-            classification,
-            count(fiche) nb_resultat,
-            avg(ind_obtention) as taux_diplomation
-        from _filtre
-        group by
-            annee_scolaire, cube (
-                school_friendly_name,
-                genre,
-                plan_interv_ehdaa,
-                population,
-                classification
-            )
-    ),
-
-    _coalesce as (
-        select
-            ind.id_indicateur,
-            ind.description_indicateur,
-            agg_dip.annee_scolaire,
-            coalesce(agg_dip.school_friendly_name, 'CSS') as ecole,
-            coalesce(agg_dip.genre, 'Tout') as genre,
-            coalesce(agg_dip.plan_interv_ehdaa, 'Tout') as plan_interv_ehdaa,
-            coalesce(agg_dip.population, 'Tout') as population,
-            coalesce(agg_dip.classification, 'Tout') as classification,
-            agg_dip.nb_resultat,
-            agg_dip.taux_diplomation
-        from agg_dip
-        inner join
-            {{ ref("pevr_dim_indicateurs") }} as ind
-            on agg_dip.id_indicateur = ind.id_indicateur
-    )
-
-select
-    id_indicateur,
-    description_indicateur,
-    annee_scolaire,
-    nb_resultat,
-    taux_diplomation,
-    {{
-        dbt_utils.generate_surrogate_key(
-            [
-                "ecole",
-                "plan_interv_ehdaa",
-                "genre",
-                "population",
-                "classification",
-            ]
-        )
-    }} as id_filtre
-from _coalesce
+{#
+Dashboards Store - Helping students, one dashboard at a time.
+Copyright (C) 2023  Sciance Inc.
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU Affero General Public License as
+published by the Free Software Foundation, either version 3 of the
+License, or any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Affero General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License
+along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#}
+
+{{ config(alias="indicateur_des") }}
+
+with
+    _mentions as (
+        select
+            mentions.fiche,
+            mentions.prog_charl,
+            (left(mentions.date_exec_sanct, 6) - 1) as brut_annee,
+            mentions.ind_reus_sanct_charl
+        from {{ ref("i_e_ri_mentions") }} as mentions
+    ),
+
+    --Création de la notion de l'année dans e_ri_mentions
+    mentions_annee as (
+        select
+            mentions.fiche,
+            mentions.prog_charl,
+            case

```

Comment:
@busquef tu peux utiliser les fonctions de date 
            case
                when month(mentions.date_exec_sanct) between 9 and 12
                then year(mentions.date_exec_sanct)
                when month(mentions.date_exec_sanct) between 1 and 8
                then (year(mentions.date_exec_sanct) - 1)
            end as annee,

Reply:
c'est déjà fait. Tu peux pull la dernière modification

Reply:
oui mais tu utilise left et right alors que c'est sous format date.. ca fait la même chose mais bon

Reply:
ahh j'ai pas remarqué.

C'est probablement moins lourd. Donc, je vais le changé pour année.

Reply:
J'utilise ce bout de code souvent :

    _mentions as (
        select
            mentions.fiche,
            mentions.prog_charl,
            (left(mentions.date_exec_sanct, 6) - 1) as brut_annee,
            mentions.ind_reus_sanct_charl
        from {{ ref("i_e_ri_mentions") }} as mentions
    ),

    _date_mentions as (
        select
            mentions.fiche,
            mentions.prog_charl,
            case
                when MONTH(brut_annee) between 9 and 12 -- Entre septembre et Décembre
                then YEAR(mentions.brut_annee)
                when MONTH(brut_annee) between 1 and 8 -- Entre Janvier et Aout
                then YEAR(mentions.brut_annee)
            end as annee,
            case
                when mentions.ind_reus_sanct_charl = 'O' then 1.0 else 0.0
            end as 'ind_obtention'
        from _mentions as mentions
    ),
    
    Penses-tu que c'est une bonne idée de faire une staging ou qqch de similaire pour réduire la réutilisation?

Reply:
ca pourrait être intéressant et faire une indice diplome pour pouvoir la réutiliser dans les autres tables de certification. ex diplome des fms ou fpt

Reply:
J'ai séparé ri_mentions et j'en ai fais une staging. J'ai updaté mes tables en conséquences

Reply:
J'ai oublié de remettre le filtre du programme charlemagne.. Je vais le corrigé avec le seqid

Reply:
Corrigé.